rpcs3/Emu/RSX/Common/ProgramStateCache.h

```diff
@@ -186,7 +190,7 @@ class program_state_cache
 	std::unordered_map <pipeline_key, pipeline_storage_type, pipeline_key_hash, pipeline_key_compare> m_storage;
 
 	std::unordered_map <pipeline_key, std::unique_ptr<async_link_task_entry>, pipeline_key_hash, pipeline_key_compare> m_link_queue;
-	std::deque<async_decompile_task_entry> m_decompile_queue;
+	lf_fifo<async_decompile_task_entry, 32> m_decompile_queue; // TODO: Determine best size
```

Comment:
Use lf_queue instead. Its mechanism of operation is different but it does not have size limitations. And yes, 32 is way too small and will fill up very quickly. Same goes for the shader cache data holder, there are games with over 10k shaders easy.

Reply:
I missed the fact lf_fifo grows indefinitely, elements are not removed when they are consumed.

lf_queue seems to fit much better but I can't find a way to iterate over the queue without consuming and removing the items. That's required to avoid adding duplicate entries to the queue. Am I missing something?